nupic/research/frameworks/vernon/distributed/experiments/supervised_experiment.py

```diff
@@ -0,0 +1,110 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import copy
+
+from torch.nn import DataParallel
+from torch.nn.parallel import DistributedDataParallel
+from torch.utils.data import DistributedSampler
+
+from nupic.research.frameworks.pytorch.distributed_sampler import (
+    UnpaddedDistributedSampler,
+)
+from nupic.research.frameworks.pytorch.model_utils import aggregate_eval_results
+from nupic.research.frameworks.vernon.distributed.experiments.components import (
+    DistributedBase,
+)
+from nupic.research.frameworks.vernon.experiments.supervised_experiment import (
+    SupervisedExperiment as SupervisedExperimentBase,
+)
+
+__all__ = [
+    "SupervisedExperiment",
+]
+
+
+class SupervisedExperiment(DistributedBase,
+                           SupervisedExperimentBase):
+    """
+    Distributed SupervisedExperiment.
+    """
+    def setup_experiment(self, config):
+        super().setup_experiment(config)
+        if self.distributed:
+            self.model = DistributedDataParallel(self.model)
+        else:
+            self.model = DataParallel(self.model)
```

Comment:
As Subutai pointed out, this might not be the most efficient approach for a single machine, although we don't know that for sure without testing. Not relevant for Imagenet but most of the experiments we have been running in Continual and MetaContinual Learning are in single instances.

It does make it more convenient but something we can keep in mind in the future is to test and see whether this is impacting single instance runs or not.